### ISSUE TITLE
Dot graph links not working with Doxygen 1.8.20

### ISSUE THREAD

**Issue by saumyajyoti** [doxygen/doxygen#7988]
**Describe the bug**
The Doxygen documentation is not navigating to linked pages from module graph, function graph when using png dot file format. svg works fine. Doxygen 1.18 worked fine.

**Expected behavior**
The referenced Doc should open when clicking a graph element.

**Screenshots**
If useful, add screenshots to help explain your problem.

**To Reproduce**
Attach a self contained example that allows us to reproduce the problem.
Such an example typically exist of some source code (can be dummy code) and a doxygen configuration file used (you can strip it using `doxygen -s -u`). After you verified the example demonstrates the problem, put it in a zip (or tarball) and attach it to the bug report. Try to avoid linking to external sources, since they might disappear in the future.

**Version**
Win32 1.8.20 

**Stack trace**
If you encounter a crash and can build doxygen from sources yourself with debug info (`-DCMAKE_BUILD_TYPE=Debug`), a stack trace can be very helpful (especially if it is not possible to capture the problem in a small example that can be shared).

**Additional context**

**Comment by albert-github** [comment]
Where does this happen?

The template says also something about:
> Attach a self contained example that allows us to reproduce the problem. ....

- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by saumyajyoti** [comment]
Will get back on this if I can repro' in a self-contained example. Currently I am having it as part of our large code base.

**Comment by albert-github** [comment]
OK. On github or is it propritory?

**Comment by albert-github** [comment]
The thing that remains is, unfortunately, to create a small example that shows the problem.

**Comment by sparist** [comment]
It seems that all diagrams produced by 1.8.20 result in non-working links, whereas 1.8.19 works fine.

This is the Doxygen CMake configuration for my project:
```
find_package(Doxygen OPTIONAL_COMPONENTS dot)
if(DOXYGEN_FOUND)
	set(DOXYGEN_BRIEF_MEMBER_DESC YES)
	set(DOXYGEN_CALL_GRAPH YES)
	set(DOXYGEN_CALLER_GRAPH YES)
	set(DOXYGEN_ENABLE_PREPROCESSING YES)
	set(DOXYGEN_EXTRACT_ALL YES)
	set(DOXYGEN_EXTRACT_ANON_NSPACES YES)
	set(DOXYGEN_EXTRACT_LOCAL_CLASSES YES)
	set(DOXYGEN_EXTRACT_PACKAGE YES)
	set(DOXYGEN_EXTRACT_PRIVATE YES)
	set(DOXYGEN_EXTRACT_STATIC YES)
	set(DOXYGEN_FULL_PATH_NAMES YES)
	set(DOXYGEN_HIDE_UNDOC_RELATIONS NO)
	set(DOXYGEN_INTERNAL_DOCS YES)
	set(DOXYGEN_JAVADOC_AUTOBRIEF YES)
	set(DOXYGEN_MACRO_EXPANSION YES)
	set(DOXYGEN_MULTILINE_CPP_IS_BRIEF YES)
	set(DOXYGEN_QT_AUTOBRIEF YES)
	set(DOXYGEN_QUIET YES)
	set(DOXYGEN_RECURSIVE YES)
	set(DOXYGEN_SORT_MEMBER_DOCS NO)
	set(DOXYGEN_STRIP_FROM_PATH targets)
	set(DOXYGEN_STRIP_FROM_INC_PATH targets/library/include)
	set(DOXYGEN_TEMPLATE_RELATIONS YES)
	set(DOXYGEN_USE_MDFILE_AS_MAINPAGE README.md)
	set(DOXYGEN_WARN_AS_ERROR YES)
	if(MSVC)
		set(DOXYGEN_WARN_FORMAT "$file($line): $text")
	endif()

	doxygen_add_docs(documentation README.md targets
		WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
		COMMENT "Generating documentation"
	)
else()
	message(STATUS "Target 'documentation' not created")
endif()
```

**Comment by saumyajyoti** [comment]
> It seems that all diagrams produced by 1.8.20 result in non-working links, whereas 1.8.19 works fine.
> 
> This is the Doxygen CMake configuration for my project:
> 
> ```
> find_package(Doxygen OPTIONAL_COMPONENTS dot)
> if(DOXYGEN_FOUND)
> 	set(DOXYGEN_BRIEF_MEMBER_DESC YES)
> 	set(DOXYGEN_CALL_GRAPH YES)
> 	set(DOXYGEN_CALLER_GRAPH YES)
> 	set(DOXYGEN_ENABLE_PREPROCESSING YES)
> 	set(DOXYGEN_EXTRACT_ALL YES)
> 	set(DOXYGEN_EXTRACT_ANON_NSPACES YES)
> 	set(DOXYGEN_EXTRACT_LOCAL_CLASSES YES)
> 	set(DOXYGEN_EXTRACT_PACKAGE YES)
> 	set(DOXYGEN_EXTRACT_PRIVATE YES)
> 	set(DOXYGEN_EXTRACT_STATIC YES)
> 	set(DOXYGEN_FULL_PATH_NAMES YES)
> 	set(DOXYGEN_HIDE_UNDOC_RELATIONS NO)
> 	set(DOXYGEN_INTERNAL_DOCS YES)
> 	set(DOXYGEN_JAVADOC_AUTOBRIEF YES)
> 	set(DOXYGEN_MACRO_EXPANSION YES)
> 	set(DOXYGEN_MULTILINE_CPP_IS_BRIEF YES)
> 	set(DOXYGEN_QT_AUTOBRIEF YES)
> 	set(DOXYGEN_QUIET YES)
> 	set(DOXYGEN_RECURSIVE YES)
> 	set(DOXYGEN_SORT_MEMBER_DOCS NO)
> 	set(DOXYGEN_STRIP_FROM_PATH targets)
> 	set(DOXYGEN_STRIP_FROM_INC_PATH targets/library/include)
> 	set(DOXYGEN_TEMPLATE_RELATIONS YES)
> 	set(DOXYGEN_USE_MDFILE_AS_MAINPAGE README.md)
> 	set(DOXYGEN_WARN_AS_ERROR YES)
> 	if(MSVC)
> 		set(DOXYGEN_WARN_FORMAT "$file($line): $text")
> 	endif()
> 
> 	doxygen_add_docs(documentation README.md targets
> 		WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
> 		COMMENT "Generating documentation"
> 	)
> else()
> 	message(STATUS "Target 'documentation' not created")
> endif()
> ```

Does it work if you force svg for image format?

**Comment by sparist** [comment]
Adding `set(DOXYGEN_DOT_IMAGE_FORMAT svg)` worked. Thanks!

**Comment by saumyajyoti** [comment]
> Adding `set(DOXYGEN_DOT_IMAGE_FORMAT svg)` worked. Thanks!

Thanks for confirming. Then it is the same issue as mine.

**Comment by albert-github** [comment]
@sparist you wrote:
> It seems that all diagrams produced by 1.8.20 result in non-working links, whereas 1.8.19 works fine.

where as @saumyajyoti wrote
> Doxygen 1.18 worked fine.

- Is the reference to 1.8.19 correct or is is it 1.8.18?
- I tried it with with code I have but here it did work as expected for png with 1.8.18, 1.8.19, 1.8.20 nd the current master. Without an example it is not possible to reproduce so (@sparist and @saumyajyoti):
   - Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by saumyajyoti** [comment]
I am able to repro the issue with below config file and Doxygen repo' source files. One more thing I discovered is that that the png issue is happening when I open in chrome browser. Internet explorer works fine. So may be some security or html out put prop changed after doxygen 1.20? 

[Doxyfile.txt](https://github.com/doxygen/doxygen/files/5313217/Doxyfile.txt)

**Comment by albert-github** [comment]
Thanks for the configuration file.

I assume you meant doxygen version 1.8.20 (you wrote 1.20).
I ran with doxygen 1.8.20 and with the current master (1.9.0 (02b8dea10813606db4a013799a3e354368d0280f)).
Problem is the fact that there are over 8500 generated images, though for the 1.8.20 version I see in call(er) graphs (in Chrome) that the links don't work, with the current master they do work. Same accounts for include dependency graphs, inheritance graphs, collaboration graphs.
(I know that some corrections have been made in this respect)

Do you have any other places where the problem occurs, when this is the case please provide the name of that page like "docs-temp/html/class_html_doc_visitor.html"

**Comment by saumyajyoti** [comment]
Thanks for the update. Yes, it is 1.8.20 is the version. Yes, same issue happens with class diagram etc. e.g. D:/github.com/docs-temp/html/class_doc_param_list.html. Good to know this is fixed in latest master.

**Comment by albert-github** [comment]
Problem has been fixed but not yet released.
It is a duplicate of #8055

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
